source/MaterialXGenShader/OCIOColorManagementSystem.cpp

```diff
@@ -0,0 +1,325 @@
+// All rights reserved.  See LICENSE.txt for license.
+//
+
+#include <MaterialXGenShader/OCIOColorManagementSystem.h>
+
+#include <MaterialXGenShader/GenContext.h>
+#include <MaterialXGenShader/ShaderGenerator.h>
+
+#include <iostream>
+
+#ifdef OpenColorIO_SKIP_IMPORTS
+#error "OCIO has been build as a non-static library"
+#endif
+#pragma warning(disable: 4275)
+#include <OpenColorIO/OpenColorIO.h>
+namespace OCIO = OCIO_NAMESPACE;
+
+namespace MaterialX
+{
+
+const string OCIOColorManagementSystem::CMS_NAME = "ocio_cms";
+const int OCIO_UNSUPPORT_LANGUAGE = -1;
+
+using OCIOLanguageMap = std::unordered_map<std::string, OCIO::GpuLanguage>;
+
+// OCIO related information
+class OCIOInformation
+{
+public:
+    // OCIO config file to use
+    OCIO::ConstConfigRcPtr config = nullptr;
+    // Mapping from supported MaterialX targets to OCIO shading languages
+    OCIOLanguageMap languageMap;
+    // Language used to code generate with (is a OCIO::GpuLanguage)
+    int language = OCIO_UNSUPPORT_LANGUAGE;
+    // MaterialX target
+    std::string target;
+};
+
+OCIOColorManagementSystemPtr OCIOColorManagementSystem::create(const string& target)
+{
+    return OCIOColorManagementSystemPtr(new OCIOColorManagementSystem(target));
+}
+
+OCIOColorManagementSystem::OCIOColorManagementSystem(const string& target) :
+    _target(target)
+{
+    _ocioInfo = new OCIOInformation();
+    _ocioInfo->config = nullptr;
+    _ocioInfo->languageMap["genglsl"] = OCIO::GPU_LANGUAGE_GLSL_4_0;
+
+    _targetSupported = false;
+}
+
+void OCIOColorManagementSystem::loadLibrary(DocumentPtr document) 
+{
+    _ocioInfo->language = OCIO_UNSUPPORT_LANGUAGE;
+
+    _document = document;
+    if (!_document)
+    {
+        throw ExceptionShaderGenError("No library loaded for color management system");
+    }
+
+    TargetDefPtr targetDef = _document->getTargetDef(_target);
+    const StringVec targets = targetDef->getMatchingTargets();
+
+    for (const string& target : targets)
+    {
+        if (_ocioInfo->languageMap.count(target))
+        {
+            _ocioInfo->language = static_cast<int>(OCIO::GPU_LANGUAGE_GLSL_4_0);
+            _ocioInfo->target = target;
+            break;                
+        }
+    }
+}
+
+bool OCIOColorManagementSystem::setConfigFile(const FilePath& configFile)
+{
+    if (_configFile != configFile && configFile.size())
+    {
+        _configFile = configFile;
+        _colorSpaceNames.clear();
+
+        _ocioInfo->config = OCIO::Config::CreateFromFile(_configFile.asString().c_str());
+        if (!_ocioInfo->config)
+        {
+            OCIO::GetCurrentConfig();
+        }
+        if (_ocioInfo->config)
+        {
+            OCIO::ColorSpaceSetRcPtr colorSpaces = _ocioInfo->config->getColorSpaces(nullptr);
+            int colorSpaceCount = colorSpaces->getNumColorSpaces();
+            for (int i = 0; i < colorSpaceCount; i++)
+            {
+                _colorSpaceNames.insert(colorSpaces->getColorSpaceNameByIndex(i));
```

Comment:
Not currently used. Just for debugging. Could be used to find available transforms ?